src/drivers/imx/sai.c

```diff
@@ -320,6 +320,17 @@ static int sai_get_fifo(struct dai *dai, int direction, int stream_id)
 	}
 }
 
+static int sai_get_hw_params(struct dai *dai,
+			     struct sof_ipc_stream_params *params,
+			     int dir)
+{
+	/* SAI only currently supports these parameters */
+	params->rate = 48000;
+	params->channels = 2;
+	params->buffer_fmt = 0;
+	params->frame_fmt = SOF_IPC_FRAME_S32_LE;
+}
+
```

Comment:
@paulstelian97 I think we should not hardcode parameters but try to read them from sai->params. Let me have a look and come up with a proposal. Hopefully today or tomorrow.

Reply:
@paulstelian97 lets go with this for now. And will fix the rest later. Since without this patch we get a crash.